server/api/src/rsFileChksum.cpp

```diff
@@ -339,11 +349,29 @@ int file_checksum(RsComm* _comm,
         }
     }};
 
-    char buffer[SVR_MD5_BUF_SZ];
+    // get the buffer size from server configuration - default to 1 MiB
+    int read_buffer_size;
+    try {
+        read_buffer_size = irods::get_advanced_setting<int>("checksum_read_buffer_size_in_bytes");
+    }
+    catch (const irods::exception&) {
+        log_api::debug("{} :: checksum_read_buffer_size_in_bytes not set. Using default of 1 MiB.", __func__);
+        read_buffer_size = 1024 * 1024;
+    }
+
+    if (read_buffer_size < 0 || read_buffer_size > INT_MAX) {
```

Comment:
If `read_buffer_size` is an int, it can never exceed `INT_MAX`.

Given that we have Boost, consider looking into the safe numerics library. See https://gcc.godbolt.org/z/aq8Gojv9z for an example.

Reply:
Should the first condition be `read_buffer_size <= 0`?

Reply:
> Should the first condition be `read_buffer_size <= 0`?

yes

Reply:
> If `read_buffer_size` is an int, it can never exceed `INT_MAX`.
> 
> Given that we have Boost, consider looking into the safe numerics library. See https://gcc.godbolt.org/z/aq8Gojv9z for an example.

I was thinking that myself since it is an int but added it because we said to have that check in there.  That said, I put a really long sequence of digits into that field in the json and it triggered the condition in 362.  Not sure how unless it somehow was parsed as a negative integer.